packages/template-retail-react-app/app/mocks/mock-data.js

```diff
@@ -2585,6 +2585,7 @@ export const mockProductSearch = {
                     id: '701644042615M'
                 }
             ],
+            variants: [],
```

Comment:
Could this empty `variants` be the one causing our tests to fail? Try filling it with some more data.

Reply:
The tests failed because it was expecting the variants to be `undefined` since there is a case where master product does not have `variants`. I've fixed the test and CI is now passed